addOns/pscanrules/src/main/java/org/zaproxy/zap/extension/pscanrules/CookieLooselyScopedScanRule.java

```diff
@@ -58,11 +63,15 @@ public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
         // name of a host from which the response has been sent from
         String host = msg.getRequestHeader().getHostName();
 
+        Set<String> ignoreList = CookieUtils.getCookieIgnoreList(getModel());
+
         // find all loosely scoped cookies
         List<HttpCookie> looselyScopedCookies = new LinkedList<HttpCookie>();
         for (HttpCookie cookie : cookies) {
             if (isLooselyScopedCookie(cookie, host)) {
-                looselyScopedCookies.add(cookie);
+                if (!ignoreList.contains(cookie.getName())) {
```

Comment:
This should be done in the above if and before calling `isLooselyScopedCookie`, avoids doing all the work if the cookie is ignored.

Reply:
Could you add the condition to the above if instead of nesting? (Same comment for the other scan rule.)

Reply:
Done.
The other rule has if-elif nested structure. I'd suggest to not merge conditionals there.

Reply:
I was suggesting adding to the if that continues, it does not add additional nesting and is straightforward to read.

Reply:
Makes sense. Updated.